### PR TITLE
[2FA] Update link msa/aad account page with legacy password account information.

### PR DESCRIPTION
**Pull request by dannyjdev**
### Changes
* Updated wording of the "Link Microsoft Account" page to include Legacy Account information.
* Legacy account page: This page will show when users have password account and are not part of the exception list.
* Link Microsoft account page: After a user with a legacy account issue sends a SR to us, we will verify the request and ownership. Once we approve it using our processes, the user will see this page and will be able to link their MSA if they know their password, otherwise they will be able to use the **Forgot your password?** link.

#### Legacy account page
Legacy Account

#### Link MSA account page
Linking account

Addresses: https://github.com/NuGet/Engineering/issues/4617

### CODE REVIEW

File: src/NuGetGallery/Views/Authentication/LinkExternal.cshtml [hunk 1]
```diff
@@ -9,29 +9,35 @@
...
-                <h1 class="text-center">Link @Model.External.ProviderAccountNoun</h1>
-            </div>
-        </div>
         <div class="row">
             <div class="@ViewHelpers.GetColumnClasses(ViewBag)">
                 @if (Model.External.ExistingUserCanBeLinked)
                 {
+                    <div class="row">
+                        <div class="col-xs-12">
+                            <h1 class="text-center">Legacy Account</h1>
```
Comment (joelverhagen) [inline src/NuGetGallery/Views/Authentication/LinkExternal.cshtml:18]: The wording seems good to me. Thoughts, @clairernovotny?

**Review by joelverhagen — CHANGES_REQUESTED** [review]
I left some comments on interaction with the allow-list.

File: src/NuGetGallery/Views/Authentication/LinkExternal.cshtml [hunk 2]
```diff
@@ -9,29 +9,35 @@
...
-                    </p>
-                    <p class="text-center">
-                        Please <a href="mailto:@Config.Current.GalleryOwner.Address">contact support</a> if you need more assistance.
+                        Some accounts can be recovered through manual proof of ownership.
+                        Please reach out to <a href="mailto:@Config.Current.GalleryOwner.Address">nuget support</a> for assistance.
                     </p>
 
-                    @Html.Partial("_SignIn", Model)
+                    <p>
+                        @Html.ShowLabelFor(m => m.SignIn.UserNameOrEmail)<br />
```
[the remark this answers is not in the document]
Comment (lyndaidaii) [reply src/NuGetGallery/Views/Authentication/LinkExternal.cshtml:30]: I think might be better show email/username in the description above?  `We found an account with this email address('email address') ....` something like this?

[the remark this answers is not in the document]
Comment (dannyjdev) [reply src/NuGetGallery/Views/Authentication/LinkExternal.cshtml]: Updated with Lynn suggestion.

File: src/NuGetGallery/Views/Authentication/LinkExternal.cshtml [hunk 3]
```diff
@@ -9,29 +9,35 @@
 <section role="main" class="container main-container page-sign-in">
     @if (Model.External.FoundExistingUser)
```
[the remark this answers is not in the document]
Comment (dannyjdev) [reply src/NuGetGallery/Views/Authentication/LinkExternal.cshtml:10]: Updated.

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**